app/models/miq_schedule_worker/runner.rb

```diff
@@ -59,58 +59,58 @@ def load_system_schedules
   end
 
   def load_user_schedules
-    return unless @active_roles.include?("scheduler")
-    sync_all_user_schedules
+    sync_all_user_schedules if schedule_enabled?(:scheduler)
   end
 
   def worker_setting_or_default(keys, default = nil)
     worker_settings.fetch_path(keys) || default
   end
 
-  def system_schedule_every(*args, &block)
-    raise ArgumentError if args.first.nil?
-    @system_scheduler.schedule_every(*args, &block)
-  rescue ArgumentError => err
-    _log.error("#{err.class} for schedule_every with #{args.inspect}.  Called from: #{caller[1]}.")
+  def schedule_enabled?(context)
+    context == :all || @active_roles.include?(context.to_s)
+  end
+
+  def scheduler_for(context)
+    @schedules[context] ||= []
+    ::MiqScheduleWorker::MiqScheduleContext.new(self, @schedules[context], @system_scheduler)
```

Comment:
Ugh, yes, I guess this is fine the way it is then.

> Can anything be done to separate that out better?

@kbrock @imtayadeway  we're only adding to the array of schedules, `@schedules`, right?  We can just return the schedules added instead.